Modules/posixmodule.c

```
@@ -9538,6 +9538,18 @@ os_getlogin_impl(PyObject *module)
     }
     else
         result = PyErr_SetFromWindowsErr(GetLastError());
+#elif defined (HAVE_GETLOGIN_R)
+    /* AFAIK the maximum length should be 32, but this is not checkable */
```

The maximum username length is 32 [on Linux](https://man7.org/linux/man-pages/man5/utmp.5.html#:~:text=UT_LINESIZE%20%20%20%20%20%2032%0A%20%20%20%20%20%20%20%20%20%20%20%23define-,UT_NAMESIZE,-32%0A%20%20%20%20%20%20%20%20%20%20%20%23define%20UT_HOSTSIZE), but you also need an extra character for the null terminator.

https://github.com/bminor/glibc/blob/d58ebf51bca1a30e1a07e800e3823f4c96029da0/sysdeps/unix/getlogin_r.c#L82

It's 255 on macOS:
https://github.com/apple-oss-distributions/xnu/blob/8d741a5de7ff4191bf97d57b9f54c2f6d4a15585/bsd/sys/param.h#L97

---

Ah, thanks!

I was looking for a constant we could use, and those links are helpful. Unfortunately it seems Linux and other *NIX systems use different ones: `UT_NAMESIZE` on Linux and `MAXLOGNAME` on the BSDs and various others. I'll update `configure` to try and detect those and fall-back to a hard-coded value of 256 if both fail.